pkg/pool-stable-phantom/contracts/StablePhantomPool.sol

```diff
@@ -232,51 +247,39 @@ contract StablePhantomPool is IRateProvider, BaseGeneralPool, ProtocolFeeCache {
         // reference them by token index in the full base tokens plus BPT set (i.e. the tokens the Pool registers). Due
         // to immutable variables requiring an explicit assignment instead of defaulting to an empty value, it is
         // simpler to create a new memory array with the values we want to assign to the immutable state variables.
-        IRateProvider[] memory tokensAndBPTRateProviders = new IRateProvider[](params.tokens.length + 1);
-        for (uint256 i = 0; i < tokensAndBPTRateProviders.length; ++i) {
+        IRateProvider[] memory tokensAndRateProviders = new IRateProvider[](params.tokens.length + 1);
```

Comment:
```suggestion
        IRateProvider[] memory allTokensRateProviders = new IRateProvider[](params.tokens.length + 1);
```

The current name is a bit weird like the array contains both token and rate provider addresses.

Reply:
It's been changing. The really long names were causing lint/readability issues.

I think the "tokens" part of it was referring to the length: like it was the size of both the tokens and rate providers. Now that we have three things, it's more confusing. Maybe just rateProviders.